src/webgpu/web_platform/copyToTexture/canvas.spec.ts

```diff
@@ -4,16 +4,58 @@ copyToTexture with HTMLCanvasElement and OffscreenCanvas sources.
 TODO: Add tests for flipY
 `;
 
+import { getResourcePath } from '../../../common/framework/resources.js';
 import { makeTestGroup } from '../../../common/framework/test_group.js';
+import { unreachable } from '../../../common/util/util.js';
 import {
   RegularTextureFormat,
   kTextureFormatInfo,
   kValidTextureFormatsForCopyE2T,
 } from '../../capability_info.js';
-import { CopyToTextureUtils, isFp16Format } from '../../util/copy_to_texture.js';
+import { CopyToTextureUtils } from '../../util/copy_to_texture.js';
 import { canvasTypes, allCanvasTypes, createCanvas } from '../../util/create_elements.js';
 
 class F extends CopyToTextureUtils {
+  init2DCanvasContentWithColorSpace({
+    width,
+    height,
+    paintOpaqueRects,
+    colorSpace,
+  }: {
+    width: number;
+    height: number;
+    paintOpaqueRects: boolean;
+    colorSpace: 'srgb' | 'display-p3';
+  }): {
+    canvas: HTMLCanvasElement | OffscreenCanvas;
+    canvasContext: CanvasRenderingContext2D | OffscreenCanvasRenderingContext2D;
+  } {
+    const canvas = createCanvas(this, 'onscreen', width, height);
+
+    let canvasContext = null;
+    canvasContext = canvas.getContext('2d', { colorSpace }) as CanvasRenderingContext2D | null;
+
+    if (canvasContext === null) {
+      this.skip('onscreen canvas 2d context not available');
+    }
+
+    if (
+      typeof (canvasContext as any).getContextAttributes === 'undefined' ||
```

Comment:
I'm using `typeof canvasContext.getContextAttributes === 'undefined'` but always got compiler error said: No property named getContextAttributes in CanvasRenderingContext2D. Do you have any suggestions?

Reply:
Both of these seem to be OK if we update tsc in package.json to 4.6.2 (latest) and run `npm install` again. There are several new typecheck errors after doing that though.